inc/saiswitch.h

```diff
@@ -60,6 +60,12 @@ typedef enum _sai_switch_oper_status_t
     /** Switch encountered a fatal error */
     SAI_SWITCH_OPER_STATUS_FAILED,
 
+    /** VOQ Switch in critical state with available fabric ports below configured configured threshold */
+    SAI_SWITCH_OPER_STATUS_VOQ_FABRIC_PORT_THRESHOLD_LOW,
+
+    /** VOQ Switch in recovery state with available fabric ports above configured configured threshold. Once the recovery is complete switch transitions to UP state */
+    SAI_SWITCH_OPER_STATUS_VOQ_FABRIC_PORT_THRESHOLD_HIGH,
```

Comment:
"fabric ports too low" is one of reasons for a switch in an error situations.  There can be other reasons, e.g. buffer or memory low, other hardware/memory errors etc.  Let's not add the error conditions here so that we can keep the switch oper status simple/clean.  thanks.

Reply:
Created a new voq switch isolate state for isolation and corresponding callback. Removed the switch over state changes.

https://github.com/opencomputeproject/SAI/pull/1793/commits/0d07fb18dc37f419ec400f7383685f1cf104fe46